tests/tests/sqlite/rawQueries.spec.ts

```diff
@@ -0,0 +1,648 @@
+              INSERT OR REPLACE INTO [User] ([id], [name], [age], [networth])
+              SELECT ?, ?, ?,
+                IFNULL((
+                  SELECT [networth] + 1000
+                  FROM [User]
+                  WHERE [id] = ?
+                ), 0)
+          `,
+            [id, name, age, id]
+          );
+
+          // ACT: Select statement to get incremented value and store it for checking later
+          const results = tx.execute('SELECT [networth] FROM [User] WHERE [id] = ?', [id]);
+
+          actual.push(results.rows?._array[0].networth);
+        });
+
+        promises.push(promised);
+      }
+
+      // ACT: Wait for all transactions to complete
+      await Promise.all(promises);
+
+      // ASSERT: That the expected values where returned
+      const expected = Array(iterations)
+        .fill(0)
+        .map((_, index) => index * 1000);
+      expect(actual).to.eql(expected, 'Each transaction should read a different value');
+    });
+
+    it('Transaction, cannot execute after commit', async () => {
+      const { id, name, age, networth } = generateUserInfo();
+      await db.writeTransaction(async (tx) => {
+        const res = tx.execute('INSERT INTO "User" (id, name, age, networth) VALUES(?, ?, ?, ?)', [
+          id,
+          name,
+          age,
+          networth
+        ]);
+
+        expect(res.rowsAffected).to.equal(1);
+        expect(res.insertId).to.equal(1);
+        expect(res.metadata).to.eql([]);
+        expect(res.rows?._array).to.eql([]);
+        expect(res.rows?.length).to.equal(0);
+        expect(res.rows?.item).to.be.a('function');
+
+        tx.commit();
+
+        let errorThrown = false;
+        try {
+          tx.execute('SELECT * FROM "User"');
+        } catch (e) {
+          errorThrown = true;
+          expect(!!e).to.equal(true);
+        }
+
+        expect(errorThrown).to.equal(true);
+      });
+
+      const res = await db.execute('SELECT * FROM User ');
+      expect(res.rows?._array).to.eql([
+        {
+          id,
+          name,
+          age,
+          networth
+        }
+      ]);
+    });
+
+    it('Incorrect transaction, manual rollback ', async () => {
+      const id = chance.string;
+      const { name, age, networth } = generateUserInfo();
+      await db.writeTransaction(async (tx) => {
+        try {
+          tx.execute('INSERT INTO "User" (id, name, age, networth) VALUES(?, ?, ?, ?)', [id, name, age, networth]);
+        } catch (e) {
+          tx.rollback();
+        }
+      });
+
+      const res = await db.execute('SELECT * FROM User');
+      expect(res.rows?._array).to.eql([]);
+    });
+
+    it('Correctly throws', async () => {
+      const id = chance.string();
+      const { name, age, networth } = generateUserInfo();
+      try {
+        await db.execute('INSERT INTO "User" (id, name, age, networth) VALUES(?, ?, ?, ?)', [id, name, age, networth]);
+      } catch (e: any) {
+        expect(!!e).to.equal(true);
```

Comment:
The test should always throw an error, due to the `id` field being a string and not a number. But you are correct, it would not cover that use case - this test was also a duplicate of `Failed insert`. I removed this tests and updated `Failed insert` to handle the case of if an error is not thrown.